From 54de2727262e9a00dc391197affcc6eca9c77d0d Mon Sep 17 00:00:00 2001
From: Mirela Rabulea <mirela.rabulea@nxp.com>
Date: Sat, 21 Jul 2018 13:26:42 +0300
Subject: [PATCH 4322/5242] MLK-18451: mxc-jpeg: Fix encoder stuck on second
 run

commit  c84eaa8e67babbc717d86a47f11580580ff747c8 from
https://source.codeaurora.org/external/imx/linux-imx.git

Do a soft reset before job run.
Encoder now runs succesivelly without crash or stuck, but outputs 0.

Signed-off-by: Mirela Rabulea <mirela.rabulea@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-jpeg.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/imx8/mxc-jpeg.c b/drivers/media/platform/imx8/mxc-jpeg.c
index efa4cee..08d4f92 100644
--- a/drivers/media/platform/imx8/mxc-jpeg.c
+++ b/drivers/media/platform/imx8/mxc-jpeg.c
@@ -367,6 +367,7 @@ static void mxc_jpeg_device_run(void *priv)
 
 	if (ctx->mode == MXC_JPEG_ENCODE) {
 		dev_dbg(dev, "Encoding on slot %d\n", slot);
+		mxc_jpeg_sw_reset(jpeg->base_reg);
 		ctx->enc_state = MXC_JPEG_ENC_CONF;
 		mxc_jpeg_config_enc(dst_buf, slot, jpeg);
 	} else {
-- 
1.7.9.5

